test/keboola/facebook/extractor/query_test.clj

```diff
@@ -67,3 +70,122 @@
+    ;; Placeholder values for unimportant parameters
+    (let [token "token"
+          id "id"
+          version "version"
+          query {}
+          ;; Parameters string as per your example
+          parameters-str "time_increment=1&breakdowns=product_id&time_ranges=[{'since':'2024-08-10','until':'2024-08-12'}]"
+          ;; Atom to capture calls to the mock function
+          async-insights-request-calls (atom [])
+          ;; Mock function to capture calls and parameters
+          mock-async-insights-request-fn (fn [token id new-parameters version query]
+                                           (swap! async-insights-request-calls conj
+                                                  {:token token
+                                                   :id id
+                                                   :parameters new-parameters
+                                                   :version version
+                                                   :query query})
+                                           ;; Return a mock result
+                                           [{:data "mock-result"}])]
+
+      ;; Call the function under test
+      (let [results (sut/make-run-inishgts-query-with-time-range
+                     mock-async-insights-request-fn
+                     token
+                     id
+                     parameters-str
+                     version
+                     query)
+            ;; Capture the calls made to the mock function
+            calls @async-insights-request-calls
+            num-calls (count calls)
+            ;; Compute expected dates based on current date
+            time-ranges-str "[{'since':'2024-08-10','until':'2024-08-12'}]"
+            time-ranges (query-parser/parse-time-ranges time-ranges-str)
+            expected-time-ranges (query-parser/expand-time-ranges time-ranges)]
+
+        ;; Verify the number of calls made matches expected number of dates
+        (is (= num-calls (count expected-time-ranges))
+            (str "Expected " (count expected-time-ranges) " calls to async-insights-request-fn, but got " num-calls))
+
+        ;; Parse the parameters once for use in generating expected parameters
+        (let [parameters-map (query-parser/parse-parameters parameters-str)]
+
+          ;; Verify each call was made with the correct parameters
+          (doseq [[call expected-time-range] (map vector calls expected-time-ranges)]
+            (let [expected-parameters (query-parser/generate-parameters-for-dates parameters-map expected-time-range)]
+              (is (= (:parameters call) expected-parameters)
+                  (str "Expected parameters: " expected-parameters ", but got: " (:parameters call))))))
+
+        ;; Optionally, verify that results contain the expected data
+        (is (= results (apply concat (repeat num-calls [{:data "mock-result"}])))
+            "Expected results to contain the mock data repeated for each call")))))
```

Comment:
testy pre query ktoru treba rozbit na mensie queries podla date_preset alebo time_ranges. Testujem pocet volani queries aj to ake time_ranges to vygenerovalo